### PR TITLE
Implement feature flagged RESTAPI payloads for Office Connector

### PR DESCRIPTION
**Pull request by Rotonen**
Amended the fixtured shadow document to have a doctype.

Made unshadowing documents trigger on file upload instead of unlocking as there was no convenient way to trip it up otherwise via the RESTAPI. Should just work as expected.

Added a feature flag for producing RESTAPI using payloads for Office Connector. The Office Connector to use this does not exist yet, but I've also implemented user story tests, including Oneoffixx tests, which show how this will work.

Closes (in tandem with the OC change): https://extranet.4teamwork.ch/support/gever-st-gallen/tracker/439

### CODE REVIEW

**Comment by Rotonen** [comment]
@buchi rebased and pending review.

**Comment by Rotonen** [comment]
@buchi rebased and amended.

File: opengever/officeconnector/interfaces.py [hunk 1]
```diff
@@ -16,6 +16,11 @@ class IOfficeConnectorSettings(Interface):
         u'OfficeConnector URLs',
         default=True)
 
+    restapi_enabled = schema.Bool(
+        title=u'OfficeConnector restapi suppport',
```
Comment (buchi) [inline opengever/officeconnector/interfaces.py]: Typo: suppport => support

**Comment by Rotonen** [comment]
Amended, rebased.

**Pull request merged**